### DISCUSSION TITLE
[Question]Pyarmor latest license: How to upgrade to the latest pyarmor license

### DISCUSSION THREAD

**Discussion by adityabaker** [dashingsoft/pyarmor#1038] [Q&A] [answered]
We are currently using pyarmor version 6.8.1 and we have an enterprise license for it. The obfuscation using the key file used to work well for us but it now gives us this error in CI:
![image](https://user-images.githubusercontent.com/92293641/228220511-df586285-186f-4681-afff-930adc293785.png)

Please note that the above error is generated when we try and run the following command to register the license:
**_pyarmor register [KEYFILE]_**

Firstly, we want to know if this problem arises because of the license. 
If yes, then we are willing to upgrade our license to the latest version available to resolve this issue. **Please let us know the procedure to upgrade our license to the latest version or purchase a new license in case upgradation is not possible.**

**Comment by jondy** [comment]
Do not upgrade to latest version or purchase new license before read this 2 pinned issues  #980 #985 
Pyarmor 8.0 has big changes, it may break current building scripts.

**Comment by jondy** [comment, marked as answer]
Pyarmor has been upgraded to 8.0, and there are big changes in Pyarmor 8.0.

If never upgrade to 8.0, there is a little change, using .zip file other than .txt file to register Pyarmor, following the second method described in the pyarmor-regcode-xxxx.txt: downloading .zip once, then use this .zip file to register later.

If could not download `.zip` file, please send this order no. to pyarmor@163.com from register email.

If you’re using CI or docker, please fix pyarmor version to 7.7.4 before making decision to upgrade Pyarmor.

Please also check this import notes for Pyarmor 8.0, do not upgrade to Pyarmor 8.0 if haven't understood what’re changed
https://github.com/dashingsoft/pyarmor/issues/980
